os/drivers/lcd/mipi_lcd.c

```diff
@@ -399,6 +406,49 @@ static int lcd_setcontrast(FAR struct lcd_dev_s *dev, unsigned int contrast)
 	return OK;
 }
 
+FAR void lcd_put_logo(FAR struct lcd_dev_s *dev)
+{
+	int logo_arr_index = 0;
+	int lcd_data_index = CONFIG_LCD_XRES * (CONFIG_LCD_YRES - LOGO_YRES) + (CONFIG_LCD_XRES - LOGO_XRES);
+	int lcd_data_col_count = 0;
+	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+	/* Memory optimization applied using Rotation buffer
+	 * If rotation is enabled, then we have two buffers allocated for rotation.
+	 * During bootup, rotation buffer will not be used (No putarea call from application)
+	 * Therefore, the rotation buffer can be safely used for storing logo data.
+	 * 
+	 * If rotation is disabled, then we need to allocate memory for full screen data
+	 * and it will allocate memory to lcd_logo_fullscreen_data buffer.
+	 * 
+	 * If rotation is enabled, then lcd_logo_fullscreen_data contains pointer of rotation buffer */
+#if defined(CONFIG_LCD_SW_ROTATION)
+	lcd_logo_fullscreen_data = lcd_buffer[0];
```

Comment:
1. maybe `lcd_buffer_index` instead of 0?
2. After flushing the log data with lcd_buffer[0], first ui data is using lcd_buffer[1] ?

Reply:
updated to lcd_buffer[1]
now first ui data will use lcd_buffer[1]